firefox-ios/Client/Frontend/Summary/MockSummarizer.swift

```diff
@@ -0,0 +1,40 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+final class MockSummarizer: SummarizerProtocol {
+    /// If set, both `summarize` and `summarizeStreamed` will throw this error.
+    var shouldThrowError: Error?
+    /// The response content, split into chunks.
+    /// Used as-is for `summarizeStreamed`, and joined with a whitespace for `summarize`.
+    var shouldRespond: [String] = []
+
+    init(shouldRespond: [String], shouldThrowError: Error?) {
+        self.shouldRespond = shouldRespond
+        self.shouldThrowError = shouldThrowError
+    }
+
+    func summarize(_ text: String) async throws -> String {
+        if let error = shouldThrowError { throw error }
+        return shouldRespond.joined(separator: " ")
+    }
+
+    func summarizeStreamed(_ text: String) -> AsyncThrowingStream<String, Error> {
+        AsyncThrowingStream { continuation in
+            Task {
```

Comment:
For these mocks, I just realize that we don't need to implement similar for production. Instead it suffices to do a check that the method was called because that's what we're really testing. However, can leave as-is since not doing harm.
 
cc: @yoanarios

Reply:
Nice. Opened a follow-up [here](https://mozilla-hub.atlassian.net/browse/FXIOS-13006) so I can clean them all up, since I have a couple of them.